src/nvme/fabrics.c

```diff
@@ -652,6 +678,62 @@ static int build_options(nvme_host_t h, nvme_ctrl_t c, char **argstr)
 		continue;		   \
 	}
 
+static bool nvme_check_default_option(const char *opt)
+{
+	int i;
+
+	for (i = 0; i < ARRAY_SIZE(default_options_strings); i++) {
+		if (!strcmp(opt, default_options_strings[i]))
+			return true;
+	}
+	return false;
+}
+
+static bool nvmf_check_option(const char *opt)
```

Comment:
This seems to almost completely duplicate `nvmf_read_options()` and `default_options`. Why can't it be implemented like this?
```c
#define nvmf_check_option(r, tok) (!nvmf_read_options(r) && r->options->tok)
```
The call site would have to change to
```c
nvmf_check_option(r, disable_sqflow)
```
This also has the benefit of caching the supported options.

Reply:
Yes, I got the same feedback on the mailing list. Let me update this accordingly.